packages/jest-worker/src/workers/WorkerAbstract.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {EventEmitter, PassThrough} from 'stream';
+import {
+  WorkerEvents,
+  WorkerInterface,
+  WorkerOptions,
+  WorkerStates,
+} from '../types';
+
+export default abstract class WorkerAbstract
+  extends EventEmitter
+  implements Pick<WorkerInterface, 'waitForWorkerReady' | 'state'>
+{
+  private __state = WorkerStates.STARTING;
```

Comment:
```suggestion
  #state = WorkerStates.STARTING;
```
instead?

Reply:
Happy with that, I just wanted a way to delineate that it shouldn't be set directly and the get/setters should be used